core/command.go

```diff
@@ -1,13 +1,57 @@
 package core
 
-import "errors"
+import (
+	"errors"
+	"sync"
+)
 
 // State represents system configuration after processing user input
 type State struct {
 	ancestors ancestors
 	Folder    *File
 	Selected  int
 	history   map[*File]int // history of all selected postions
+	*FileMarker
+}
+
+type FileMarker struct {
+	Marked []*File
```

Comment:
You can use `map` here,  I was earlier suggesting `map[*File]string` where the string would be path computed from `State.ancestors`, but if you want to use it only as a set, you can use map[*File]struct{}.

Than you can add new `*File` to the map by `Makred[newMarkedFile] = struct{}` and figure out if file is marked by `_, isMarked = Marked[fileToCheck]`